### DISCUSSION TITLE
Hover and click events of component doesn't work in a book

### DISCUSSION THREAD

**Discussion by ScarletRedMan** [Minestom/Minestom#1330] [Help] [answered]
Hello. I've a problem with hover and click events in a book. I made it with method `Component#hoverEvent(Component.text("..."))` but it doesn't working for using in a book, but it's working in a chat. 
```java
Component someComponent = Component.text("Some text")
    .hoverEvent(Component.text("One more some text"))
    .clickEvent(ClickEvent.openUrl("http://192.168.0.13/"));

Book book = Book.book(
    Component.text("Some book"), 
    Component.text("nameless"), 
    List.of(someComponent)
);
player.openBook(book);
```

**Comment by KrystilizeNevaDies** [comment]
I have written a fix for this, however I'm not going to pr it just yet.
https://pastes.dev/Fg2tn7Ymu9

**Comment by KrystilizeNevaDies** [comment, marked as answer]
Fixed by [https://github.com/Minestom/Minestom/pull/1359](url)
